File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
let cnt = 0;
function solve(node) {
    cnt *= 10;
    cnt += node.val;
    if (node.left === null && node.right === null)
        return cnt;
    let res = 0;
    if (node.left !== null) {
        res += solve(node.left);
        cnt = parseInt(cnt / 10);
    }
    if (node.right !== null) {
        res += solve(node.right);
        cnt = parseInt(cnt / 10);
    }

    return res;
} 

var sumNumbers = function(root) {
    cnt = 0;
    return solve(root);
};

File: 0129-sum-root-to-leaf-numbers/0129-sum-root-to-leaf-numbers.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    
    int solve(TreeNode* node,int val)
    {
        if (!node -> left && !node -> right)
            return (val * 10) + node -> val;
        
        int res = 0;
        if (node -> left)
            res += solve(node -> left, (val * 10) + node -> val);
        if (node -> right)
            res += solve(node -> right, (val * 10) + node -> val);
        return res;
    }
    
    int sumNumbers(TreeNode* root) {
        return solve(root,0);
    }
};